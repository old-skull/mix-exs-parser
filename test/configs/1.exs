defmodule App.MixProject do
  use Mix.Project
  use Mix.Env

  defp description do
    """
    Multiline comment as description of the application
    """
  end

  defp extra_applications(_), do: [:logger]
  defp extra_applications(:test), do: [:stream_data | extra_applications(123)]

  def project do
    [
      app: :app,
      version: "0.1.0",
      elixir: "~> 1.13",
      start_permanent: Mix.env() == :prod,
      deps: deps()
    ]
  end

  def application do
    # Run "mix help compile.app" to learn about applications.
    [
      extra_applications: [:logger]
    ]
  end

  # Run "mix help deps" to learn about dependencies.
  defp deps do
    [
      # {:dep_from_hexpm, "~> 0.3.0"},
      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
    ]
  end
end
