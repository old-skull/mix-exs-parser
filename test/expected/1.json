{
  "moduleName": "App.MixProject",
  "uses": [
    "Mix.Project"
  ],
  "comments": {
    "single": [
      "# Run \"mix help compile.app\" to learn about applications.",
      "# Run \"mix help deps\" to learn about dependencies.",
      "# {:dep_from_hexpm, \"~> 0.3.0\"},",
      "# {:dep_from_git, git: \"https://github.com/elixir-lang/my_dep.git\", tag: \"0.1.0\"}"
    ],
    "multi": []
  },
  "defs": {
    "project": {
      "rawContent": "[\n      app: :app,\n      version: \"0.1.0\",\n      elixir: \"~> 1.13\",\n      start_permanent: Mix.env() == :prod,\n      deps: deps()\n    ]",
      "content": {
        "app": ":app",
        "version": "0.1.0",
        "elixir": "~>1.13",
        "start_permanent": "Mix.env()==:prod",
        "deps": "deps()"
      },
      "isPrivate": false
    },
    "application": {
      "rawContent": "[\n      extra_applications: [:logger]\n    ]",
      "content": {
        "extra_applications": ":logger"
      },
      "isPrivate": false
    },
    "deps": {
      "rawContent": "[\n      # {:dep_from_hexpm, \"~> 0.3.0\"},\n      # {:dep_from_git, git: \"https://github.com/elixir-lang/my_dep.git\", tag: \"0.1.0\"}\n    ]",
      "content": {},
      "isPrivate": true
    }
  }
}